Web Inspector: Search in console doesn't work when Inspector is <820px wide

stwrt opened the pull request:
#### 5c25fec73760ab96dc3594a896293afbd5676423
<pre>
Web Inspector: Search in console doesn&apos;t work when Inspector is &lt;820px wide
<a href="https://bugs.webkit.org/show_bug.cgi?id=198016">https://bugs.webkit.org/show_bug.cgi?id=198016</a>
<a href="https://rdar.apple.com/problem/50922509">rdar://problem/50922509</a>

Reviewed by Devin Rousso.

The search bar is hidden in the Console tab when the web inspector window gets too small.
This is currently the only search bar to demonstrate this behavior from what I can tell.
The network filter and the global search both display even at the smallest size.

The margin-inline-end size of 3px is designed to give consistent spacing between elements and ensure
that the filter and search bar have a proper gap.

* Source/WebInspectorUI/UserInterface/Views/LogContentView.css:
(.navigation-bar &gt; .item.find-banner.console):
* Source/WebInspectorUI/UserInterface/Views/LogContentView.js:
(WI.LogContentView):

Canonical link: <a href="https://commits.webkit.org/304057@main">https://commits.webkit.org/304057@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4271b8b3c833cb3f978b0dc53b901a5b6ee53497

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134424 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/6888 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/45644 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/141957 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/86408 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a2ea3194-7ac0-4ed9-b554-40a1a9da9f92) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7487 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/6754 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/102745 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/70002 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ad5c511b-79e8-4849-83c1-5a42971b40f9) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137371 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5224 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120472 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83536 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e67a3d7b-0b40-406b-90a4-277a09449a4f) 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/133773 "Build is in progress. Recent messages:") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5084 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/2702 "Passed tests") | [⏳ 🛠 wpe-cairo-libwebrtc ](https://ews-build.webkit.org/#/builders/WPE-Cairo-LibWebRTC-Build-EWS "Waiting in queue, processing has not started yet") | | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/126478 "Build is in progress. Recent messages:") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/114292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/38613 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144647 "Built successfully") | | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/132932 "Build is in progress. Recent messages:") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6568 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39197 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/111150 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/6644 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5512 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111420 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28264 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4917 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/116749 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60362 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6620 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/34947 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6443 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/70191 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/6679 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6554 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/LogContentView.css:
```diff
@@ -63,6 +63,11 @@
+.navigation-bar > .item.find-banner.console:has(+ .log-scope-bar.force-hidden + .divider.force-hidden + .message-channel-scope-bar.force-hidden),
+.navigation-bar > .item.find-banner.console:has(+ .log-scope-bar.force-hidden + .console-other-filters-button) {
+    padding-inline-end: 3px;
+}
```
can you attach some screenshots of before and after for this?  im not sure i really understand why this is necessary (or if maybe there's a different approach that is less complex)

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/LogContentView.js:
```diff
@@ -61,7 +61,7 @@ WI.LogContentView = class LogContentView extends WI.ContentView
 
         this._findBanner = new WI.FindBanner(this, {alwaysShowing: true, className: "console"});
-        this._findBanner.visibilityPriority = WI.NavigationItem.VisibilityPriority.Low;
+        this._findBanner.visibilityPriority = WI.NavigationItem.VisibilityPriority.High;
```
we probably should adjust this for some of the other navigation items to make sure that we have the right priorities (or really just change that entire system to not be number based)

but that can be a followup :)

stwrt committed:
Web Inspector: Search in console doesn't work when Inspector is <820px wide
https://bugs.webkit.org/show_bug.cgi?id=198016
rdar://problem/50922509

Reviewed by Devin Rousso.

The search bar is hidden in the Console tab when the web inspector window gets too small.
This is currently the only search bar to demonstrate this behavior from what I can tell.
The network filter and the global search both display even at the smallest size.

The margin-inline-end size of 3px is designed to give consistent spacing between elements and ensure
that the filter and search bar have a proper gap.

* Source/WebInspectorUI/UserInterface/Views/LogContentView.css:
(.navigation-bar > .item.find-banner.console):
* Source/WebInspectorUI/UserInterface/Views/LogContentView.js:
(WI.LogContentView):

Canonical link: https://commits.webkit.org/304057@main
